sdk/key_vault/src/key.rs

```diff
@@ -296,6 +344,52 @@ impl<'a, T: TokenCredential> KeyClient<'a, T> {
         result.algorithm = algorithm;
         Ok(result)
     }
+
+    /// Decrypt a single block of encrypted data.
+    /// The DECRYPT operation decrypts a well-formed block of ciphertext using the target encryption key and specified algorithm.
+    /// This operation is the reverse of the ENCRYPT operation; only a single block of data may be decrypted, the size of this block is dependent on the target key and the algorithm to be used.
+    /// The DECRYPT operation applies to asymmetric and symmetric keys stored in Vault or HSM since it uses the private portion of the key. This operation requires the keys/decrypt permission.
+    pub async fn decrypt(
+        &mut self,
+        algorithm: EncryptionAlgorithm,
+        key_name: &str,
+        key_version: &str,
+        value: &str,
+        aad: Option<&str>,
+        iv: Option<&str>,
+        tag: Option<&str>,
+    ) -> Result<DecryptResult, Error> {
+        // POST {vaultBaseUrl}/keys/{key-name}/{key-version}/decrypt?api-version=7.2
+
+        let mut uri = self.vault_url.clone();
+        uri.set_path(&format!("keys/{}/{}/decrypt", key_name, key_version));
+        uri.set_query(Some(API_VERSION_PARAM));
+
+        let mut request_body = Map::new();
+        request_body.insert("alg".to_owned(), Value::String(algorithm.to_string()));
+        request_body.insert("value".to_owned(), Value::String(value.to_owned()));
+
+        if let Some(aad) = aad {
```

Comment:
These members should be `Vec<u8>` (or `Option<Vec<u8>>`) and base64url-encoded here - don't make the user do it in their own code, which is inconsistent with our other Azure SDKs.